File: front/kaemeweb/src/app/modules/shared/services/fornecedor.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable, tap } from 'rxjs';
import { Cliente } from '../models/cliente';
import { Fornecedor } from '../models/fornecedor';

@Injectable({
  providedIn: 'root'
})
export class FornecedorService {

  constructor(private httpClient: HttpClient) { }

  create(fornecedor: Fornecedor): Observable<boolean> {
    return this.httpClient.post<boolean>('https://localhost:7072' + '/fornecedor/Create', fornecedor);
  }

  update(fornecedor: Fornecedor): Observable<boolean> {
    return this.httpClient.put<boolean>('https://localhost:7072' + '/fornecedor/Update', fornecedor);
  }

  delete(id: number): Observable<boolean> {
    let params = new HttpParams()
    params = params.append('id', id)

    return this.httpClient.delete<boolean>('https://localhost:7072' + '/fornecedor/Delete', {params});
  }

  getAll(): Observable<Fornecedor[]> {
    return this.httpClient.get<Fornecedor[]>('https://localhost:7072' + '/fornecedor/GetAll');
  }

  getFornecedorById(id: number): Observable<Fornecedor> {
    let params = new HttpParams()
    params = params.append('id', id)

    return this.httpClient.get<Fornecedor>('https://localhost:7072' + '/fornecedor/GetFornecedorById', {params});
  }

}

File: back/KaemeWebApi/Dominio/Services/FornecedorService.cs
using Dominio.Interface;
using Dominio.Models;
using Shared.RepositoryShared;

namespace Dominio.Services
{
    public class FornecedorService
    {
        private IFornecedorRepository _fornecedorRepository;

        public FornecedorService(IFornecedorRepository fornecedorRepository)
        {
            _fornecedorRepository = fornecedorRepository;
        }

        public RepositoryResult Create(Fornecedor fornecedor)
        {
            try
            {
                var result = _fornecedorRepository.Create(fornecedor);
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }

        public RepositoryResult Update(Fornecedor fornecedor)
        {
            try
            {
                var result = _fornecedorRepository.Update(fornecedor);
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }

        public RepositoryResult Delete(int id)
        {
            try
            {
                var result = _fornecedorRepository.Delete(id);
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }         

        public RepositoryResult GetAll()
        {
            try
            {
                var result = _fornecedorRepository.GetAll();
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }

        public RepositoryResult GetFornecedorById(int id)
        {
            try
            {
                var result = _fornecedorRepository.GetFornecedorById(id);
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }        

    }
}